src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/4.sql

```diff
@@ -753,6 +760,9 @@ AS
     DECLARE @seriesKey BIGINT
     DECLARE @instanceKey BIGINT
 
+    IF @extendedQueryTagsVersion <> (SELECT MAX(TagVersion) FROM dbo.ExtendedQueryTag)
```

Comment:
How is AddExtendedQueryTag locking on the max watermark it needs to go re-index?

Depending on that you should test out all concurrent combinations to make sure this is enough

Reply:
Will add `holdlock`  on it.